contracts/modules/STO/POLYCappedSTO.sol

```diff
@@ -0,0 +1,526 @@
+
+    /**
+     * @dev Modifies STO start and end times
+     * @param _startTime start time of sto
+     * @param _endTime end time of sto
+     */
+    function modifyTimes(uint256 _startTime, uint256 _endTime) external onlyOwner notStarted {
+        _modifyTimes(_startTime, _endTime);
+    }
+
+    /**
+     * @dev Modifies addresses used as wallet and treasury wallet
+     * @param _wallet Address of wallet where funds are sent
+     */
+    function modifyWalletAddress(address payable _wallet) external onlyOwner {
+        //can be modified even when STO started
+        _modifyWalletAddress(_wallet);
+    }
+
+    /**
+     * @notice Use to change the treasury wallet
+     * @param _treasuryWallet Ethereum account address to receive unsold tokens
+     */
+    function modifyTreasuryWallet(address _treasuryWallet) external onlyOwner {
+        emit SetTreasuryWallet(treasuryWallet, _treasuryWallet);
+        treasuryWallet = _treasuryWallet;
+    }
+
+    function _modifyLimits(
+        uint256 _minimumInvestment,
+        uint256 _nonAccreditedLimit,
+        uint256 _maxNonAccreditedInvestors
+    ) internal {
+        minimumInvestment = _minimumInvestment;
+        nonAccreditedLimit = _nonAccreditedLimit;
+        maxNonAccreditedInvestors = _maxNonAccreditedInvestors;
+        emit SetLimits(minimumInvestment, nonAccreditedLimit, maxNonAccreditedInvestors);
+    }
+
+    function _modifyRate(uint256 _rate) internal {
+        require(_rate > 0, "Rate not > 0");
+        rate = _rate;
+        emit SetRate (rate);
+    }
+
+    function _modifyCap(uint256 _cap) internal {
+        require(_cap > 0, "Cap not > 0");
+        cap = _cap;
+        emit SetCap (cap);
+    }
+
+    function _modifyTimes(uint256 _startTime, uint256 _endTime) internal {
+        /*solium-disable-next-line security/no-block-members*/
+        require(_startTime >= now && _endTime > _startTime, "Invalid times");
+        startTime = _startTime;
+        endTime = _endTime;
+        emit SetTimes(_startTime, _endTime);
+    }
+
+    function _modifyWalletAddress(address payable _wallet) internal {
+        require(_wallet != address(0), "Invalid wallet");
+        emit SetWallet(wallet, _wallet);
+        wallet = _wallet;
+    }
+
+    ////////////////////
+    // STO Management //
+    ////////////////////
+
+    /**
+     * @notice Finalizes the STO and mints remaining tokens to treasury address
+     * @notice Treasury address must be whitelisted to successfully finalize
+     * @param mintUnsoldTokens unsold tokens will be minted to the Treasury wallet if ture
+     */
+    function finalize(bool mintUnsoldTokens) external onlyOwner {
+        require(!isFinalized, "Already finalized");
+        isFinalized = true;
+        if ((mintUnsoldTokens) && (totalTokensSold < cap)) {
+//            address _wallet = (treasuryWallet == address(0) ? IDataStore(getDataStore()).getAddress(TREASURY) : treasuryWallet);
+            address _wallet = treasuryWallet;
```

Comment:
No need for `_wallet`

Reply:
_wallet is here as a place holder. the line above is what I was going to replace it with when "Add treasury wallet in global storage" is merged to dev-3.0.0.